### PR TITLE
 Fix minimum_degree_ordering

### PR DESCRIPTION
**Pull request by CromwellEnage**
Handle degenerate case where input graph contains no edges.

### CODE REVIEW

File: test/Jamfile.v2 [hunk 1]
```diff
@@ -144,6 +130,35 @@ test-suite graph_test :
...
+    [ run random_spanning_tree_test.cpp ../build//boost_graph ]
+    [ run mas_test.cpp ../../test/build//boost_unit_test_framework/<link>static : $(TEST_DIR) ]
+    [ run stoer_wagner_test.cpp ../../test/build//boost_unit_test_framework/<link>static : $(TEST_DIR) ]
+    ;
+
+test-suite graph_test :
+    graph_test_regular
+    graph_test_no_msvc14
+    ;
+
```
Comment (jeremy-murphy) [inline test/Jamfile.v2]: Why all these other changes that seem unrelated to `minimum_degree_ordering`? Maybe better to put them in a different PR?

Comment (jeremy-murphy) [reply test/Jamfile.v2]: Oh, I get it now, you merged your parameter fix in here too. Looks like we should get that one merged asap.

**Comment by jzmaddock** [comment]
Can you rebase this to develop please?

**Comment by CromwellEnage** [comment]
Rebased.

File: include/boost/graph/minimum_degree_ordering.hpp [hunk 2]
```diff
@@ -317,7 +319,8 @@ namespace boost {
         // Initialize degreelists.  Degreelists organizes the nodes
         // according to their degree.
         for (boost::tie(v, vend) = vertices(G); v != vend; ++v) {
-          put(degree, *v, out_degree(*v, G));
+          put(degree, *v, d = out_degree(*v, G));
+          if (0 < d) has_no_edges = false;
           degreelists.push(*v);
         }
```
Comment (jeremy-murphy) [inline include/boost/graph/minimum_degree_ordering.hpp]: For simplicity, I would prefer the assignment `d = ...` to happen before the `put` rather than inline, and I would move the declaration of `d` inside that loop too. It's just a preference for scope and simplicity, nothing serious.

**Comment by jeremy-murphy** [comment]
Looks good to me!

**Comment by CromwellEnage** [comment]
@jzmaddock, your verdict?

**Comment by jzmaddock** [comment]
Yep, all good I think, merging.
